testing/src/test/java/org/oppia/testing/FakeExceptionLoggerTest.kt

```diff
@@ -0,0 +1,194 @@
+package org.oppia.testing
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.util.logging.ExceptionLogger
+import org.robolectric.annotation.Config
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class FakeExceptionLoggerTest {
+
+  @Inject
+  lateinit var fakeExceptionLogger: FakeExceptionLogger
+
+  @Inject
+  lateinit var exceptionLogger: ExceptionLogger
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testFakeExceptionLogger_logException_returnsException() {
+    exceptionLogger.logException(IllegalStateException("Test Exception"))
+    val exception = fakeExceptionLogger.getMostRecentException()
+
+    assertThat(exception).isInstanceOf(IllegalStateException::class.java)
+    assertThat(exception).hasMessageThat().contains("Test Exception")
+  }
+
+  @Test
+  fun testFakeExceptionLogger_logExceptionTwice_returnsLatestException() {
+    exceptionLogger.logException(IllegalStateException("First Exception"))
+    exceptionLogger.logException(NullPointerException("Second Exception"))
+    val exception = fakeExceptionLogger.getMostRecentException()
+
+    assertThat(exception).isInstanceOf(NullPointerException::class.java)
+    assertThat(exception).hasMessageThat().contains("Second Exception")
+  }
+
+  @Test
+  fun testFakeExceptionLogger_logException_clearList_logExceptionAgain_returnsLatestException() {
+    exceptionLogger.logException(IllegalStateException("First Exception"))
+    fakeExceptionLogger.clearAllExceptions()
+    exceptionLogger.logException(NullPointerException("Second Exception"))
+    val exception = fakeExceptionLogger.getMostRecentException()
+
+    assertThat(exception).isInstanceOf(NullPointerException::class.java)
+    assertThat(exception).hasMessageThat().contains("Second Exception")
+  }
+
+  @Test
+  fun testFakeExceptionLogger_logNothing_getMostRecent_returnsFailure() {
+    val exception = assertThrows(NoSuchElementException::class) {
+      fakeExceptionLogger.getMostRecentException()
+    }
+
+    assertThat(exception).isInstanceOf(NoSuchElementException::class.java)
+    assertThat(exception).hasMessageThat().contains("List is empty.")
```

Comment:
This is technically an implementation detail of Kotlin we can probably omit. It's a good idea to check the message of the failure, but in this case the context and exception is pretty specific, so maybe just keep the check for ``NoSuchElementException`` type.

Reply:
Done.